bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/evaluation/storage/EvaluationStorage.kt

```diff
@@ -0,0 +1,19 @@
+package io.bucketeer.sdk.android.internal.evaluation.storage
+
+import io.bucketeer.sdk.android.internal.model.Evaluation
+
+internal interface EvaluationStorage {
+  val userId: String
+  var currentEvaluationsId: String
+  var featureTag: String
+  var userAttributesUpdated: Boolean
+
+  // expected set evaluatedAt from `deleteAllAndInsert` only
+  val evaluatedAt: String
+
+  fun getBy(featureId: String): Evaluation?
+  fun get(): List<Evaluation>
```

Comment:
How do you distinguish which should be a property and which should be a method?

If you just decide it by its underlying implementation, then we should just use methods for better consistency.

Reply:
thank for reviewing @yshrsmz 

The decision stems from two primary reasons:

- Consistency in Implementation: I have chosen to maintain the same implementation approach as before when we did on the previous `EvaluationDao` & `EvaluationInteractor` 

- Simplicity and Reduced Overhead: These functions serve primarily for 'get' and 'set' purposes. By creating separate methods for each of them, the interface would has  numerous methods dedicated solely to 'get' and 'set' operations. Instead, I have opted to encapsulate the 'get' and 'set' logic within the property definition itself, taking advantage of Kotlin's ability to override property accessors."

What do you think @yshrsmz

Reply:
One more thing , I think I could improvement here is move logic `set` evaluation_id to the `deleteAllAndInsert` & `update` as they should done together

Reply:
I am worried that future maintainers may not understand its difference and could use property accessors for SQLite-related APIs, which is unacceptable.

> set | update evaluation_id to the deleteAllAndInsert & update as they should done together

This sounds good to me. We should reduce unnecessary APIs 👍

Reply:
updated in https://github.com/bucketeer-io/android-client-sdk/pull/89/commits/eaf30f23b79316ecca7a2bc46635c6099da416a8

Reply:
Hi @yshrsmz , I pushed new changes 
- [x] Remove almost properties from the `EvaluationStorage`
- [x] Move `set evaluation_id` to the `deleteAllAndInsert` & `update` method